aws-organizations-account/src/main/java/software/amazon/organizations/account/BaseHandlerStd.java

```diff
@@ -183,25 +185,33 @@ public final ProgressEvent<ResourceModel, CallbackContext> handleRetriableExcept
         final AccountConstants.Action actionName,
         final AccountConstants.Handler handlerName
     ) {
-        try {
-            String accountInfo = model.getAccountId() == null ? handlerRequest.getLogicalResourceIdentifier() : model.getAccountId();
-            if (actionName != AccountConstants.Action.CREATE_ACCOUNT) {
-                int currentAttempt = context.getCurrentRetryAttempt(actionName, handlerName);
-                if (currentAttempt < MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION) {
+        String accountInfo = model.getAccountId() == null ? handlerRequest.getLogicalResourceIdentifier() : model.getAccountId();
+
+
+        if (actionName != AccountConstants.Action.CREATE_ACCOUNT) {
+            int currentAttempt = context.getCurrentRetryAttempt(actionName, handlerName);
+            if (currentAttempt < MAX_RETRY_ATTEMPT_FOR_RETRIABLE_EXCEPTION) {
+                context.setCurrentRetryAttempt(actionName, handlerName);
+                if (actionName == AccountConstants.Action.LIST_PARENTS
+                        || actionName == AccountConstants.Action.LIST_TAGS_FOR_RESOURCE
```

Comment:
Could we do this by handler? If this block is meant to just catch retryable exceptions for Read, List we could just use, for example, Handler.READ

Reply:
+1 to this, is there a reason why we are listing specific APIs rather than using the Handler type?

Reply:
Updated the PR to use handler name instead of action name.